hazelcast/src/main/java/com/hazelcast/collection/impl/collection/CollectionService.java

```diff
@@ -90,8 +88,12 @@ public void shutdown(boolean terminate) {
 
     @Override
     public void destroyDistributedObject(String name) {
-        getContainerMap().remove(name);
-        nodeEngine.getEventService().deregisterAllListeners(getServiceName(), name);
+        CollectionContainer container = getContainerMap().remove(name);
+        if (container != null) {
+            container.clear(false);
```

Comment:
Do the extra clear statement here, offer sth more than attempt to help GC?

Reply:
Without it the backup testing fails (which might need a later cleanup).

Reply:
I've replaced it with a `container.destroy()` call and added the missing `clear()` on the backups in the `SetContainer.onDestroy()` method. This should now properly destroy the container, including the transaction and backup data.